LanguageFeatures/Patterns/switch_statement_A04_t01.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We extend switch statements to allow patterns in cases:
+///
+/// switchStatement         ::= 'switch' '(' expression ')'
+///                         '{' switchStatementCase* switchStatementDefault? '}'
+/// switchStatementCase     ::= label* 'case' guardedPattern ':' statements
+/// guardedPattern          ::= pattern ( 'when' expression )?
+/// switchStatementDefault  ::= label* 'default' ':' statements
+/// ...
+/// Many constant expressions are subsumed by the new pattern syntax so most
+/// existing switch cases have the same semantics under this proposal. However,
+/// patterns are not a strict superset of constant expressions and some switches
+/// may be broken.
+/// ...
+/// Constant constructors. A switch case can be a constant constructor call with
+/// implicit const, like:
+/// ```
+/// case SomeClass(1, 2):
+/// ```
+/// With this proposal, that is interpreted as an object pattern whose arguments
+/// are subpatterns.
+///
+/// @description Checks that constant constructor is not called in case of an
+/// object pattern in a switch statement
```

Comment:
It seems that in either case, the const constructor and the object pattern, the test will pass. I suggest using a constant expression in the field initializers to make the field values different from the constructor arguments, for example, `this.value = value + 1`.

Reply:
That makes sense. Did the change